piker/brokers/kraken.py

```diff
@@ -616,42 +617,41 @@ async def handle_order_requests(
                 reqid=msg.reqid
             )
 
-            try:
-                # Check to make sure there was no error returned by
-                # the kraken endpoint. Assert one order was cancelled
-                assert resp['error'] == []
-                assert resp['result']['count'] == 1
-
-                # TODO: Change this code using .get
-                try:
-                    pending = resp['result']['pending']
-                # Check to make sure the cancellation is NOT pending,
-                # then send the confirmation to the ems order stream
-                except KeyError:
-                    await ems_order_stream.send(
-                        BrokerdStatus(
-                            reqid=msg.reqid,
-                            account=msg.account,
-                            time_ns=time.time_ns(),
-                            status='cancelled',
-                            reason='Order cancelled',
-                            broker_details={'name': 'kraken'}
-                        ).dict()
-                    )
-            except AssertionError:
-                log.error(f'Order cancel was not successful')
+            # Check to make sure there was no error returned by
+            # the kraken endpoint. Assert one order was cancelled
+            assert resp['error'] == []
+            assert resp['result']['count'] == 1
+
+            # Check to make sure the cancellation is NOT pending,
+            # then send the confirmation to the ems order stream
+            pending = resp['result'].get('pending')
```

Comment:
please be sure this logic looks right, it's not clear to me at all why you assert things then catch the error around another block that has nothing to do with the assert. this is just generally non-pythonic to catch and handle asserts; the whole point is they raise.

Reply:
makes sense. Is this something you want to address in this PR?

Reply:
I don't really ever like seeing `except AssertionError:` unless in some kinda wonky testing (and even then).

If you think this works then i'm fine to land it - makes the code a lot more readable for me.